frappe/public/js/frappe/form/toolbar.js

```diff
@@ -52,10 +52,14 @@ frappe.ui.form.Toolbar = Class.extend({
 		this.set_indicator();
 	},
 	is_title_editable: function() {
-		if (this.frm.meta.title_field==="title"
+		let title_field = this.frm.meta.title_field;
+		let doc_field = this.frm.get_docfield(title_field);
+
+		if (title_field
 			&& this.frm.perm[0].write
-			&& !this.frm.get_docfield("title").options
-			&& !this.frm.doc.__islocal) {
+			&& !this.frm.doc.__islocal
+			&& !doc_field.options
```

Comment:
Should explicitly check for fieldtype **Data** here. Options can be set for fieldtype **Data**.

Reply:
noted